rules/regions.smk

```diff
@@ -0,0 +1,222 @@
+            --group-by {params.group_by} \
+            --sequences-per-group {params.sequences_per_group} \
+            --output {output.sequences} \
+        """
+
+rule make_priorities:
+    message:
+        """
+        determine priority for inclusion in as phylogenetic context by
+        genetic similiarity to sequences in focal set for region '{wildcards.region}'.
+        """
+    input:
+        alignment = rules.mask.output.alignment,
+        metadata = rules.download.output.metadata,
+        focal_alignment = rules.subsample_focus.output.sequences
+    output:
+        priorities = REGION_PATH + "subsampling_priorities.tsv"
+    resources:
+        mem_mb = 4000
+    conda: "../envs/nextstrain.yaml"
+    shell:
+        """
+        python3 scripts/priorities.py --alignment {input.alignment} \
+            --metadata {input.metadata} \
+            --focal-alignment {input.focal_alignment} \
+            --output {output.priorities}
+        """
+
+rule subsample_context:
+    message:
+        """
+        Subsample the non-focal sequences to provide phylogenetic context for the region '{wildcards.region}' using {params.sequences_per_group} per {params.group_by}.
+        """
+    input:
+        sequences = rules.mask.output.alignment,
+        metadata = rules.download.output.metadata,
+        priorities = rules.make_priorities.output.priorities
+    output:
+        sequences = REGION_PATH + "subsample_context.fasta"
+    params:
+        group_by = config["subsample_context"]["group_by"],
+        sequences_per_group = config["subsample_context"]["sequences_per_group"]
+    conda: "../envs/nextstrain.yaml"
+    shell:
+        """
+        augur filter \
+            --exclude-where region={wildcards.region} \
+            --sequences {input.sequences} \
+            --metadata {input.metadata} \
+            --priority {input.priorities} \
+            --group-by {params.group_by} \
+            --sequences-per-group {params.sequences_per_group} \
+            --output {output.sequences}
+        """
+
+rule subsample_regions:
+    message:
+        """
+        Combine and deduplicate FASTAs
+        """
+    input:
+        rules.subsample_focus.output.sequences,
+        rules.subsample_context.output.sequences
+    output:
+        alignment = REGION_PATH + "subsampled_alignment.fasta"
+    conda: "../envs/nextstrain.yaml"
+    shell:
+        """
+        python3 scripts/combine-and-dedup-fastas.py \
+            --input {input} \
+            --output {output}
+        """
+
+rule adjust_metadata_regions:
+    message:
+        """
+        Adjusting metadata for region '{wildcards.region}'
+        """
+    input:
+        metadata = rules.download.output.metadata
+    output:
+        metadata = REGION_PATH + "metadata_adjusted.tsv"
+    conda: "../envs/nextstrain.yaml"
+    shell:
+        """
+        python3 scripts/adjust_regional_meta.py \
+            --region "$region" \

```

Comment:
This expansion doesn't work for me, and seems to be expecting a bash variable `$region` to exist. I changed it to the following for testing 
```
    params:
        region = "{region}"
    conda: "../envs/nextstrain.yaml"
    shell:
        """
        python3 scripts/adjust_regional_meta.py \
            --region {params.region} \
```

Reply:
Good catch! I fixed this with the wildcard value.